kv/range_cache.go

```diff
@@ -149,7 +162,10 @@ func (rmc *RangeDescriptorCache) EvictCachedRangeDescriptor(key proto.Key) {
 // the cache.
 func (rmc *RangeDescriptorCache) getCachedRangeDescriptor(key proto.Key) (
 	rangeCacheKey, *proto.RangeDescriptor) {
-	metaKey := engine.RangeMetaKey(key)
+	// We want to look up the range descriptor for key. The cache is
+	// indexed using the end-key of the range, but the end-key is
+	// non-inclusive. So we access the cache using key.Next().
```

Comment:
I'd guess that this code was based on the same idea as the `meta{1,2}` indexes. Not sure why end-key is used there as it causes similar problems in `InternalRangeLookup`.